src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -205,15 +213,29 @@ public async Task<IActionResult> ExecuteAsync(StoredProcedureRequestContext cont
 
             // Open connection and execute query using _queryExecutor
             string queryString = queryBuilder.Build(structure);
-            JsonDocument? jsonDocument =
-                await queryExecutor.ExecuteQueryAsync(
-                    sqltext: queryString,
-                    parameters: structure.Parameters,
-                    dataReaderHandler: queryExecutor.GetJsonResultAsync<JsonDocument>,
-                    httpContext: _httpContextAccessor.HttpContext!,
-                    args: null,
-                    dataSourceName: dataSourceName);
+
+            if (await _featureManager.IsEnabledAsync("CachingPreview"))
+            {
+                if (_runtimeConfigProvider.GetConfig().SqlDataSourceUsed
```

Comment:
Have we created a separate issue to track caching for CosmosDB GraphQL?

Reply:
#1884